common/src/test/java/org/apache/uniffle/common/ReconfigurableRegistryTest.java

```diff
@@ -0,0 +1,139 @@
+    ReconfigurableBad bad = new ReconfigurableBad();
+    ReconfigurableGood good0 = new ReconfigurableGood();
+    ReconfigurableGood good1 = new ReconfigurableGood();
+    ReconfigurableGood good1Follow = new ReconfigurableGood();
+    ReconfigurableGood goodAny = new ReconfigurableGood();
+    ReconfigurableGood good2 = new ReconfigurableGood();
+    ReconfigurableGood good01 = new ReconfigurableGood();
+    try {
+      ReconfigurableRegistry.register("key0", good0);
+      ReconfigurableRegistry.register(bad);
+      ReconfigurableRegistry.register("key1", good1);
+      ReconfigurableRegistry.register("key1", good1Follow);
+      ReconfigurableRegistry.register(goodAny);
+      ReconfigurableRegistry.register("key2", good2);
+      ReconfigurableRegistry.register(Sets.newHashSet("key0", "key1"), good01);
+      ReconfigurableRegistry.update(null, Collections.singletonMap("key", "value"));
+
+      assertEquals(0, good0.mInvokeCount);
+      assertEquals(0, good1.mInvokeCount);
+      assertEquals(0, good1Follow.mInvokeCount);
+      assertEquals(1, goodAny.mInvokeCount);
+      assertEquals(0, good2.mInvokeCount);
+      assertEquals(0, good01.mInvokeCount);
+
+      ReconfigurableRegistry.update(null, Collections.singletonMap("key1", "value1"));
+
+      assertEquals(0, good0.mInvokeCount);
+      assertEquals(1, good1.mInvokeCount);
+      assertEquals(1, good1Follow.mInvokeCount);
+      assertEquals(2, goodAny.mInvokeCount);
+      assertEquals(0, good2.mInvokeCount);
+      assertEquals(1, good01.mInvokeCount);
+
+      Map<String, Object> changedProperties = new HashMap<>();
+      changedProperties.put("key0", "value0");
+      changedProperties.put("key1", "value1");
+      changedProperties.put("key2", "value2");
+      ReconfigurableRegistry.update(null, changedProperties);
+
+      assertEquals(1, good0.mInvokeCount);
+      assertEquals(2, good1.mInvokeCount);
+      assertEquals(2, good1Follow.mInvokeCount);
+      assertEquals(5, goodAny.mInvokeCount);
+      assertEquals(1, good2.mInvokeCount);
+      assertEquals(3, good01.mInvokeCount);
+    } finally {
+      assertTrue(ReconfigurableRegistry.unregister(bad));
+      assertTrue(ReconfigurableRegistry.unregister(Sets.newHashSet("key0"), good0));
+      assertTrue(ReconfigurableRegistry.unregister(Sets.newHashSet("key1")));
+      assertTrue(ReconfigurableRegistry.unregister(goodAny));
+      assertTrue(ReconfigurableRegistry.unregister("key2", good2));
+      assertTrue(ReconfigurableRegistry.unregister(Sets.newHashSet("key0", "key1"), good01));
+      assertEquals(0, ReconfigurableRegistry.getSize());
+    }
+  }
+
+  class ReconfigurableBad implements ReconfigurableRegistry.ReconfigureListener {
+    @Override
+    public void update(RssConf conf, Map<String, Object> changedProperties) {
+      throw new RuntimeException("I am bad guy");
+    }
+  }
+
+  class ReconfigurableGood implements ReconfigurableRegistry.ReconfigureListener {
```

Comment:
Do we still need this?

Reply:
Yes, I need a ReconfigurableGood#mInvokeCount to verify whether the result is expected.